File: java/Leaf under budget.java
class Solution{
    public int getCount(Node node, int bud)
    {
        //code here  
        Queue<Pair> q = new LinkedList<>();
        q.add(new Pair(node,1));
        int count = 0;
        int sum = 0;
        
        while(q.size()>0){
            Node temp = q.peek().a;
            int lvl = q.peek().b;
            q.remove();
            if(temp.left==null && temp.right==null){
                count++;
                sum = sum + lvl;
                if(sum>bud) return count-1;
                if(sum==bud) return count;
            }
            if(temp.left!=null) q.add(new Pair(temp.left,lvl+1));
            if(temp.right!=null) q.add(new Pair(temp.right,lvl+1));
        }
        return count;
        
    }
}

class Pair{
    Node a;
    int b;
    Pair(Node a, int b){
        this.a = a;
        this.b = b;
    }
}

File: cpp/Leaf under budget.cpp
class Solution
{
public:
    int getCount(Node *root, int k)
    {
       
        int res=0;
        int sum=0;
        queue<pair<Node*,int>>q;
        q.push({root,1});
        while(!q.empty())        {
            Node* temp=q.front().first;
            int lvl=q.front().second;
            q.pop();
            
            if(temp->left==NULL && temp->right==NULL)
            {
                res++;
                sum += lvl;
                if(sum==k)return res;
                if(sum>k)return res-1; 
            }
            if(temp->left)q.push({temp->left,lvl+1});
            if(temp->right)q.push({temp->right,lvl+1});
        }

        return res;
    }
};